### PR TITLE
Fix Filter Bar Filter Markup and CSS Classes; closes #2592.

### LINKED ISSUE #2592: Fix Filter Bar Filter Markup and CSS Classes
Tickets List Filter Bar Anomolies

Here's the outer markup for the dateslist filter bar:

```html
<div class="ee-filter-bar__collapsible">

	<div class="ee-filter-bar__filter ee-dates-datetimes-to-show-filter">
		<div class="ee-base-input">
			<label class="ee-base-input-label">status</label>
			<div class="ant-select ant-select-single ant-select-show-arrow"></div>
		</div>
	</div>
	
	<div class="ee-filter-bar__filter ee-dates-datetimes-to-show-filter">
		<div class="ee-base-input">
			<label class="ee-base-input-label">sales</label>
			<div class="ant-select ant-select-single ant-select-show-arrow"></div>
		</div>
	</div>
	
	<div class="ee-filter-bar__select">
		<div class="ee-base-input espresso-date-list-filter-bar-display-select">
			<label class="ee-base-input-label">display</label>
			<div class="ant-select ant-select-single ant-select-show-arrow"></div>
		</div>
	</div>

	<div class="ee-filter-bar__select">
		<div class="ee-base-input espresso-date-list-filter-bar-order-select">
			<label class="ee-base-input-label">sort by</label>
			<div class="ant-select ant-select-single ant-select-show-arrow"></div>
		</div>
	</div>
	
	<div class="ee-base-input ee-filter-bar__filter">
		<label class="ee-base-input-label" for="ee-search-input-dates-list">search</label>
		<input id="ee-search-input-dates-list" class="ant-input ee-entity-list-filter-bar-search" type="text" value="">
	</div>

</div>
```

and here's the outer markup for the tickets list filter bar

```html
<div class="ee-filter-bar__collapsible">

	<div class="ee-filter-bar__filter ee-dates-datetimes-to-show-filter">
		<div class="ee-base-input">
			<label class="ee-base-input-label">status</label>
			<div class="ant-select ant-select-single ant-select-show-arrow"></div>
		</div>
	</div>


	<div class="ee-ticket-dates-chained-filter ee-filter-bar__filter ee-filter-bar__filter--micro">
	<label class="esprs-button-label screen-reader-text" for="ee-ticket-list-filter-bar-is-chained">Only Show Tickets for Dates Above</label>
	<button id="ee-ticket-list-filter-bar-is-chained" tabindex="0" type="button" class="ant-btn ee-filter-bar-filter-btn ee-ticket-list-not-chained esprs-button esprs-btn-minimal ant-btn-icon-only ant-btn-icon-only"></button>
	</div>

	<div class="ee-filter-bar__filter ee-dates-datetimes-to-show-filter">
		<div class="ee-base-input">
			<label class="ee-base-input-label">sales</label
			><div class="ant-select ant-select-single ant-select-show-arrow"></div>
		</div>
	</div>

	<div class="ee-tickets-display-start-or-end-ticket-filter ee-filter-bar-filter">
		<div class="ee-base-input espresso-date-list-filter-bar-display-select">
			<label class="ee-base-input-label">display</label>
			<div class="ant-select ant-select-single ant-select-show-arrow"></div>
		</div>
	</div>

	<div class="ee-tickets-sort-by-filter ee-filter-bar-filter">
		<div class="ee-base-input espresso-ticket-list-filter-bar-order-select">
			<label class="ee-base-input-label">sort by</label>
			<div class="ant-select ant-select-single ant-select-show-arrow"></div>
		</div>
	</div>div>

	<div class="ee-base-input ee-filter-bar__filter">
		<label class="ee-base-input-label" for="ee-search-input-tickets-list">search</label>
		<input id="ee-search-input-tickets-list" class="ant-input ee-entity-list-filter-bar-search" type="text" value="">
	</div>

</div>
```

and here's a least a few of the problems:

- there's **four** elements using the `ee-dates-datetimes-to-show-filter` class ?!?!?! :laughing:

- classes for the input wrapper divs is inconsistent, with `ee-filter-bar__filter`, `ee-filter-bar__select`, and `ee-filter-bar-filter`. They should all use the same class block and element name

- the outer divs for the "search" filters use the `ee-base-input` class, whereas all other filters use that on an inner div. The structural markup should be consistent for all filters.

- the "search" filter labels (and hidden isChained  label) are the only ones that implement the `for` attribute, which is important for a11y. IDs should be added for all inputs and `for` attributes added to each label using the corresponding ID accordingly.

- the CSS that was previously applied to the isChained button is missing. I'm guessing it's a result of the recent sass changes in that file. The nesting is likely off in the sass file but should be corrected.

### PR DESCRIPTION
**Pull request by nicolad**
## Problem this Pull Request solves
- there's **four** elements using the `ee-dates-datetimes-to-show-filter` class ?!?!?! :laughing:

- classes for the input wrapper divs is inconsistent, with `ee-filter-bar__filter`, `ee-filter-bar__select`, and `ee-filter-bar-filter`. They should all use the same class block and element name

- the outer divs for the "search" filters use the `ee-base-input` class, whereas all other filters use that on an inner div. The structural markup should be consistent for all filters.

- the "search" filter labels (and hidden isChained  label) are the only ones that implement the `for` attribute, which is important for a11y. IDs should be added for all inputs and `for` attributes added to each label using the corresponding ID accordingly.

- the CSS that was previously applied to the isChained button is missing. I'm guessing it's a result of the recent sass changes in that file. The nesting is likely off in the sass file but should be corrected.

- Closes: Fix Filter Bar Filter Markup and CSS Classes #2592

### CODE REVIEW

File: assets/src/domain/eventEditor/ui/datetimes/hooks/useDatesListFilterBar.tsx [hunk 1]
```diff
@@ -23,7 +23,7 @@ const useDatesListFilterBar = (): DatesListFilterBarCallback => {
 		registerFilterBarItem('status', ({ filterState }) => {
 			const { status, setStatus } = filterState;
 			return (
-				<div className='ee-filter-bar__filter ee-dates-datetimes-to-show-filter'>
```
Comment (nicolad) [inline assets/src/domain/eventEditor/ui/datetimes/hooks/useDatesListFilterBar.tsx]: `ee-dates-datetimes-to-show-filter` was obsolete

File: assets/src/domain/eventEditor/ui/tickets/ticketsList/filterBar/controls/TicketsChainedButton.tsx [hunk 2]
```diff
@@ -16,18 +15,18 @@ interface TicketsChainedButtonProps {
...
 		? __('tickets list is linked to dates list and is showing tickets for above dates only')
 		: __('tickets list is unlinked and is showing tickets for all event dates');
-	const className = isChained
-		? 'ee-filter-bar-filter-btn ee-active-filters ee-ticket-list-is-chained'
-		: 'ee-filter-bar-filter-btn ee-ticket-list-not-chained';
-	const icon = isChained ? Icon.LINK : Icon.UNLINK;
+
 	return (
 		<>
-			<label className='esprs-button-label screen-reader-text' htmlFor={'ee-ticket-list-filter-bar-is-chained'}>
```
Comment (nicolad) [inline assets/src/domain/eventEditor/ui/tickets/ticketsList/filterBar/controls/TicketsChainedButton.tsx]: based on this article https://webaim.org/techniques/forms/controls#button:
`For form buttons (submit and reset input elements and button elements), no additional accessibility information is required.`

Comment (nicolad) [reply assets/src/domain/eventEditor/ui/tickets/ticketsList/filterBar/controls/TicketsChainedButton.tsx]: when a button have text inside that will be used for a11y

Comment (nicolad) [reply assets/src/domain/eventEditor/ui/tickets/ticketsList/filterBar/controls/TicketsChainedButton.tsx]: but when there is no text inside, like in this component it needs to have `aria-label`, this is why I refactored a bit `EspressoButton` to check the `buttonText` and `tooltip`(which is acting like a label)

Comment (nicolad) [reply assets/src/domain/eventEditor/ui/tickets/ticketsList/filterBar/controls/TicketsChainedButton.tsx]: `aria-label` was used based on this recommendation: https://developers.google.com/web/tools/lighthouse/audits/button-name

File: assets/src/domain/eventEditor/ui/tickets/hooks/useTicketsListFilterBar.tsx [hunk 3]
```diff
@@ -33,7 +33,7 @@ const useTicketsListFilterBar = (): TicketsListFilterBarCallback => {
 		registerFilterBarItem('isChained', ({ filterState }) => {
 			const { isChained, toggleIsChained } = filterState;
 			return (
-				<div className='ee-ticket-dates-chained-filter ee-filter-bar__filter ee-filter-bar-filter--micro'>
+				<div className='ee-filter-bar__chain ee-filter-bar__filter ee-filter-bar__filter--micro'>
```
Comment (tn3rb) [inline assets/src/domain/eventEditor/ui/tickets/hooks/useTicketsListFilterBar.tsx]: 👍

File: assets/src/domain/eventEditor/ui/tickets/ticketsList/filterBar/controls/TicketsChainedButton.tsx [hunk 4]
```diff
@@ -16,18 +15,18 @@ interface TicketsChainedButtonProps {
  * will appear, otherwise ALL tickets will appear (subject to other filters)
  */
 const TicketsChainedButton: React.FC<TicketsChainedButtonProps> = ({ isChained, toggleIsChained }) => {
+	const className = classNames('ee-filter-bar__btn', {
+		'ee-filter-bar__btn--active': isChained,
+	});
```
Comment (tn3rb) [inline assets/src/domain/eventEditor/ui/tickets/ticketsList/filterBar/controls/TicketsChainedButton.tsx]: 👍

File: assets/src/application/ui/layout/entityList/filterBar/EntityListFilterBar.tsx [hunk 5]
```diff
@@ -52,12 +52,9 @@ const EntityListFilterBar = <FS extends ELFSM>({
...
 				{filerBarItems}
-				<SearchInput
-					className={'ee-filter-bar__filter'}
-					id={listId}
-					searchText={searchText}
-					setSearchText={setSearchText}
-				/>
+				<div className='ee-filter-bar__filter'>
+					<SearchInput id={listId} searchText={searchText} setSearchText={setSearchText} />
+				</div>
```
Comment (tn3rb) [inline assets/src/application/ui/layout/entityList/filterBar/EntityListFilterBar.tsx]: 👍

**Pull request merged**